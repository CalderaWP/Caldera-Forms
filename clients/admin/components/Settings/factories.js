import deepmerge from 'deepmerge';

/**
 * Provides CF Pro Local settings, with defaults merged to stored values.
 *
 * @param {Object} stored Stored values
 */
export const proLocalSettingsFactory = (stored = {} ) => {
	const defaults = {
		connected: false,
		generalSettings : {
			enhancedDelivery: false,
			generatePDFs: false,
			logLevel: 200
		},
		apiKeys: {
			public: '',
			secret: '',
			token: ''
		}

	};
	return deepmerge(
		defaults,
		stored,
	);


};