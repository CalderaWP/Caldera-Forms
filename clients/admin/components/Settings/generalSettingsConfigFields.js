import {optionFactory} from "../util/optionFactory";

export const STYLE_FORM = 'form';
export const STYLE_GRID = 'grid';
export const STYLE_ALERT = 'alert';
export const CDN_ENABLE = 'cdn';
const enableOption = optionFactory(
	true,
	'Enable'
);
export default  [
	{
		id: 'form',
		label: 'Form Styles',
		desc: 'Includes Bootstrap 3 styles on the frontend for form fields and buttons',
		type: 'fieldset',
		inputType: 'checkbox',
		path: 'generalSettings.form',
		options: [enableOption]
	},
	{
		id: 'alert',
		label: 'Alert Styles',
		desc: 'Includes Bootstrap 3 styles on the frontend for form alert notices',
		type: 'fieldset',
		inputType: 'checkbox',
		path: 'stylgeneralSettingseIncludes.alert',
		options: [enableOption]
	},
	{
		id: 'grid',
		label: 'Grid',
		desc: 'Includes Bootstrap 3 styles on the frontend for form grid layouts',
		type: 'fieldset',
		inputType: 'checkbox',
		path: 'generalSettings.grid',
		options: [enableOption]
	},{
		id: 'cdn',
		label: 'Enable Free CDN',
		desc: 'Some usage data will be shared with CDN providers',
		type: 'fieldset',
		innerType: 'checkbox',
		path: 'generalSettings.cdnEnable',
		options: [enableOption]
	},

]