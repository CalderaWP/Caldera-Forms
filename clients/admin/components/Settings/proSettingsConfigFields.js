import {optionFactory} from "../util/optionFactory";

export const PUBLIC_KEY = 'proPublicKey';
export const PRIVATE_KEY = 'proPrivateKey';
export const ENHANCED_DELIVERY = 'enhancedDelivery';
export const LOG_LEVEL = 'logLevel';
export const GENEREATE_PDFS = 'generatePDFs';
export const PRO_API_KEYS = 'apiKeys';
export const PRO_GENERAL_SETTINGS = 'generalSettings';
export const PRO_FORM_SETTINGS = 'formSettings';


export default {

	apiKeys: [
		{
			id: PUBLIC_KEY,
			label: 'Public Key',
			desc: 'Public Key For Caldera FormsSlot Pro',
			type: 'text',
			path: 'apiKeys.public',
			default: '',
		},
		{
			id: PRIVATE_KEY,
			label: 'Private Key',
			desc: 'Private Key For Caldera FormsSlot Pro',
			path: 'apiKeys.secret',
			type: 'text',
			options: [
				{
					label: 'yes',
					value: 'on'
				}
			],
			default: '',
		},
	],
	generalSettings: [
		{
			id: ENHANCED_DELIVERY,
			label: 'Enable Enhanced Delivery',
			desc: 'If enabled, messages are sent via Caldera FormsSlot Pro Servers',
			type: 'checkbox',
			path: 'generalSettings.enhancedDelivery',
			default: true,
		},
		{
			id: LOG_LEVEL,
			label: 'Minimum Log Level',
			desc: 'Setting a higher level than notice may affect performance, and should only be used when instructed by support.',
			type: 'select',
			path: 'generalSettings.logLevel',
			options: [
				optionFactory(
					100,
					'DEBUG'
				),
				optionFactory(
					200,
					'INFO'
				),
				optionFactory(
					300,
					'WARNING'
				),
				optionFactory(
					400,
					'ERROR'
				),
				optionFactory(
					500,
					'ALERT'
				),
				optionFactory(
					600,
					'EMERGENCY'
				),
			],
			default: 400,
		},
	],
	[PRO_FORM_SETTINGS]: [

	]
};





