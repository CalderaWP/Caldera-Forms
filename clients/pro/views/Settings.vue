<template>
	<div id="cf-pro-message-settings">
		<div id="cf-pro-message-setting-inner">
			<div class="caldera-editor-header">
				<ul class="caldera-editor-header-nav">
					<li class="caldera-editor-logo">
						<span class="caldera-forms-name">
							Caldera Forms Pro
						</span>
					</li>
					<li class="status good" v-if="connected">
						Connected
					</li>
					<li class="status bad" v-if="!connected">
						Not Connected
					</li>
					<li class="cf-pro-save">
						<input type="submit" class="button button-primary" value="Save" @click="save"/>
					</li>
					<li id="cf-pro-alert-wrap">
						<status
							:message="mainAlert.message"
							:success="mainAlert.success"
							:show="mainAlert.show"
							>
						</status>
					</li>
				</ul>
			</div>
			<div class="cf-pro-settings" v-cloak>
				<div>
					<tabs :options="{ useUrlFragment: false }">
						<tab name="Account">
							<account-edit></account-edit>
						</tab>
						<tab name="Form Settings">
							<div v-if="connected">
								<forms-settings></forms-settings>
							</div>
							<div v-else>
								You must connected to Caldera Forms Pro First
							</div>
						</tab>
						<tab name="Settings">
							<delivery></delivery>
							<logs></logs>
						</tab>
					</tabs>
				</div>
			</div>
		</div>
	</div>
</template>
<script>
	import { mapState, mapGetters, mapActions, mapMutations } from 'vuex';
	import AccountDisplay from '../components/Account/display';
	import AccountEdit from '../components/Account/Edit';
	import FormsSettings from '../components/FormSettings/Forms';
	import enhancedDelivery from '../components/GeneralSettings/enhancedDelivery';
    import logLevel from '../components/GeneralSettings/logLevel';
	import Status from '../components/Elements/Status/Status.vue'
	export default{
		components :{
			'account-display': AccountDisplay,
			'account-edit' : AccountEdit,
			'forms-settings' : FormsSettings,
			'delivery' : enhancedDelivery,
            'logs' : logLevel,
			'status' : Status
		},
		computed: mapState({
			loading: state => state.loading,
			connected: state => state.connected,
			publicKey: state => state.account.apiKeys.public,
			enhancedDelivery: state => state.settings.enhancedDelivery,
            logLevel: state => state.settings.logLevel,
			mainAlert: state => state.mainAlert
		}),
		beforeMount(){
			[].forEach.call(document.querySelectorAll('.update-nag'),function(e){
				e.parentNode.removeChild(e);
			});

			this.$store.dispatch( 'getAccount' );
		},
		methods:{
			save(){
				this.$store.dispatch( 'saveAccount' );
			}
		}
	}
</script>
<style>
	[v-cloak] {
		display: none;
		visibility: hidden;
	}

	.cf-pro-settings {
		margin-top: 50px;
	}

	.cf-pro-save input.button {
		margin: 10px;
	}

	li.status {
		padding: 14px !important;
		color: white;
	}

	li.status.good {
		background-color: #0b7a6f;
	}

	li.status.bad {
		background-color: red;
	}


	.tabs-component-panels,ul.tabs-component-tabs {
		float: left;
		display:inline-block
	}

	.tabs-component-panels{
		padding: 1rem;
		font-size: 1rem;
	}
	ul.tabs-component-tabs{
		background: #0b7a6f;

	}
	li.tabs-component-tab {
		margin-bottom: 0;
		border-bottom: thin solid white;
		padding: 0;
	}
	li.tabs-component-tab:last-child() {
		border-bottom: none;
	 }
	li.tabs-component-tab a{
		display: block;
		width: 69%;
		color: white;
		font-size: 1rem;
		padding: 1rem 1.4rem;
		text-decoration: none;
		text-align: center;
	}
	li.tabs-component-tab.is-active{
		background-color: #ff7e30;
	}

	.tabs-component-panels section:nth-child(3) .caldera-config-group {
		margin: 0 1rem 0;
	}
	.tabs-component-panels section:nth-child(3) label {
		display: inline;
		margin: 0;
	}

</style>