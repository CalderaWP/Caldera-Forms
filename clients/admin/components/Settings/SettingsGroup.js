import React from 'react';
import PropTypes from 'prop-types';
import {object, pick} from "dot-object";
import {RenderGroup} from '@caldera-labs/components';
import {Button} from '@wordpress/components';
import classNames from 'classnames'
export class SettingsGroup extends React.PureComponent{

	constructor(props) {
		super(props);
		this.state = {
			[props.settingsKey]: props.settings
		};
		this.onSettingsChange = this.onSettingsChange.bind(this);
		this.getConfigFields = this.getConfigFields.bind(this);
		this.onSettingsSave = this.onSettingsSave.bind(this);
		this.wrapperClass = this.wrapperClass.bind(this);

	}

	/**
	 * Save the settings
	 */
	onSettingsSave() {
		this.props.onSettingsSave(this.state[this.props.settingsKey]);
	}

	/**
	 * Update internal state when settings change
	 *
	 * @param {Object} update
	 */
	onSettingsChange(update) {
		this.setState(
			{
				[this.props.settingsKey]:update
		});
	}

	/**
	 * Prepare config fields
	 *
	 * @return {Array}
	 */
	getConfigFields() {
		let currentConfigFields = this.props.configFields;
		currentConfigFields.forEach(configField => {
			const {path} = configField;
			if( undefined !== path ){
				configField.value = pick(
					path,
					this.state
				);
				configField.onValueChange = (newValue) => {
					const update = {
						...this.state,
						[path]:newValue
					};
					this.onSettingsChange(object(update));
				};
			}





		});
		return currentConfigFields;
	};

	/**
	 * Get the class for the outer element
	 * @return {String}
	 */
	wrapperClass(){
		return this.props.wrapperClass;
	}


	render(){
		if( Array.isArray( this.props.configFields ) ) {
			return(
				<div
					className={classNames(
						this.wrapperClass(),
						this.props.classNames
					)}
				>
					<RenderGroup
						configFields={this.getConfigFields()}
					/>
					<Button
						onClick={this.props.onSettingsSave}
					>
						{this.props.saveButtonText}
					</Button>
				</div>
			)
		}
		return <div>!</div>

	}

}

/**
 * Prop types for the SettingsGroup component
 * @type {{}}
 */
SettingsGroup.propTypes = {
	classNames: PropTypes.string,
	settings: PropTypes.object.isRequired,
	settingsKey: PropTypes.string.isRequired,
	onSettingsSave: PropTypes.func.isRequired,
	configFields: PropTypes.array,
	saveButtonText: PropTypes.string,
};

SettingsGroup.defaultProps= {
	saveButtonText: 'Save'
}