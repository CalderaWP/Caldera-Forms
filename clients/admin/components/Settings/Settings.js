import React from 'react';
import classNames from 'classnames'
import {TabPanel} from '@wordpress/components';
import proSettingsConfigFields from './proSettingsConfigFields';
import generalSettingsConfigFields from  './generalSettingsConfigFields';
import {SettingsGroup} from "./SettingsGroup";
import Grid from 'react-css-grid';
import {ProWhatIs} from "./ProSettings/ProWhatIs/ProWhatIs";
import {ProFreeTrial} from "./ProSettings/ProFreeTrial/ProFreeTrial";
import {ProEnterApp} from "./ProSettings/ProEnterApp/ProEnterApp";
import PropTypes from 'prop-types';

/**
 * Creates the UI for Caldera FormsSlot global settings
 */
export class Settings extends React.PureComponent {

	/**
	 * Create Settings componet
	 * @param props
	 */
	constructor(props) {
		super(props);
		this.onSettingsSave = this.onSettingsSave.bind(this);
		this.getConfigFields = this.getConfigFields.bind(this);
	}


	/**
	 * Dispatches settings to parent on save
	 * @param update
	 */
	onSettingsSave(update) {
		this.props.updateSettings(update);
	};


	getConfigFields(tabName){
		switch(tabName){
			case 'generalSettings':
				return generalSettingsConfigFields;
			case 'apiKeys':
				return proSettingsConfigFields.apiKeys;
			case 'proGeneral':
				return proSettingsConfigFields.generalSettings;
			default:
				return [];
		}
	}

	/**
	 * Creat main Caldera FormsSlot settings UI
	 * @return {*}
	 */
	render() {
		const {settings,onSettingsSave} = this.props;
		return (
			<div>
				<TabPanel
					orientation={'horizontal'}
					className={
						classNames(Settings.classNames.wrapper, this.props.classNames)
					}
					activeClass={
						classNames(Settings.classNames.active)
					}
					tabs={[
						{
							name: 'generalSettings',
							title: 'Global Form Settings',
						},
						{
							name: 'apiKeys',
							title: 'Api Keys',
							className: 'privacy-settings',
						},
						{
							name: 'proGeneral',
							title: 'Caldera Forms Pro',
							className: 'pro-general-settings',
						}
					]}
				>
					{
						(tabName) => {
							return <SettingsGroup
								settings={settings}
								settingsKey={tabName}
								onSettingsSave={onSettingsSave}
								configFields={this.getConfigFields(tabName)}
							/>
						}
					}
				</TabPanel>
				<Grid>
					{this.props.proConnected &&
						<React.Fragment>
							<ProWhatIs/>
							<ProFreeTrial/>

						</React.Fragment>
					}
					{this.props.proConnected &&
						<React.Fragment>
							<ProEnterApp/>
						</React.Fragment>
					}


				</Grid>
			</div>

		);
	}
};

Settings.propTypes = {
	generalSettings: PropTypes.shape({
		form: PropTypes.boolean,
		grid:  PropTypes.boolean,
		alert:  PropTypes.boolean,
		cdn:  PropTypes.boolean
	}),
	proSettings: PropTypes.shape({
		connected: PropTypes.boolean,
		apiKeys:  PropTypes.shape({
			proPublicKey: PropTypes.string,
			proPrivateKey: PropTypes.string,
		}),
		proGeneralSettings: PropTypes.shape({
			enhancedDelivery: PropTypes.boolean,
			logLevel: PropTypes.string
		}),
	})
};



/**
 * Class names used in the Settings component
 * @type {{wrapper: string}}
 */
Settings.classNames = {
	wrapper: 'caldera-forms-global-settings',
	active: 'caldera-forms-settings-tab-active'
};