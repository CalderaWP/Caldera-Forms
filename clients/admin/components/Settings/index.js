import {proLocalSettingsFactory} from "./factories";
import {Settings} from "./Settings";
import {SettingsGroup} from "./SettingsGroup";

/**
 *
 */
export default {
	proLocalSettingsFactory,
	Settings,
	SettingsGroup
}