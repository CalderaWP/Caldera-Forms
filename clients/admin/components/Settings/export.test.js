import Settings from './index';
import {snapshotObjectKeysAndTypes} from "../../testUtil/snapshotObjectKeysAndTypes";

describe( 'Settings export', () => {
	it( 'index has the right types and keys', () => {
		snapshotObjectKeysAndTypes(Settings);
	});
});