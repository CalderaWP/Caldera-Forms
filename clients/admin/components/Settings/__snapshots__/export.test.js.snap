// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings export index has the right types and keys 1`] = `"{\\"proLocalSettingsFactory\\":\\"function\\",\\"Settings\\":\\"function\\",\\"SettingsGroup\\":\\"function\\"}"`;
